### PR TITLE
User Story 60501: Azure Analysis Services

### PR DESCRIPTION
**Pull request by TomArcherMsft**
Part of POC to test generating sample code and articles using OpenAI.

### CODE REVIEW

File: quickstart/101-analysis-services-create/variables.tf [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+  description = "Location for all resources."
+}
+
+variable "resource_group_name_prefix" {
+  default     = "rg"
+  description = "Prefix of the resource group name that's combined with a random ID so name is unique in your Azure subscription."
+}
+
+variable "server_name" {
+  description = "The name of the Azure Analysis Services server to create. Server name must begin with a letter, be lowercase alphanumeric, and between 3 and 63 characters in length. Server name must be unique per region."
```
Comment (lonegunmanb) [inline quickstart/101-analysis-services-create/variables.tf]: Our e2e test cannot work with `variable` without default value, could you please add a default value for this? Thanks!

File: quickstart/101-analysis-services-create/TestRecord.md [hunk 2]
```diff
@@ -0,0 +1 @@
+Article not yet tested by automated testing pipeline.
```
Comment (lonegunmanb) [inline quickstart/101-analysis-services-create/TestRecord.md]: You don't need this empty TestRecord.md file, the pipeline would generate one once this pr got merged.

File: quickstart/101-analysis-services-create/main.tf [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+  name     = random_pet.rg_name.id
+  location = var.resource_group_location
+}
+
+resource "azurerm_analysis_services_server" "server" {
+  name                = var.server_name
+  resource_group_name = azurerm_resource_group.rg.name
+  location            = azurerm_resource_group.rg.location
+  sku                 = var.sku
+  backup_blob_container_uri = var.backup_blob_container_uri
```
Comment (lonegunmanb) [inline quickstart/101-analysis-services-create/main.tf:22]: Would you please run `terraform fmt` for this folder to align all equal symbols? Thanks.

File: quickstart/101-analysis-services-create/providers.tf [hunk 4]
```diff
@@ -0,0 +1,16 @@
+terraform {
+  required_version = ">=0.12"
+  required_providers {
+    azurerm = {
+      source  = "hashicorp/azurerm"
+      version = "~>2.0"
```
Comment (lonegunmanb) [inline quickstart/101-analysis-services-create/providers.tf]: As now azurerm's major version is 3.0, should we set `version = "~> 3.0"` here?

**Review by lonegunmanb — CHANGES_REQUESTED** [review]
Thanks @TomArcherMsft for opening pr, we've got some issues to solve.

File: quickstart/101-analysis-services-create/TestRecord.md (hunk 2 above)
Comment (TomArcherMsft) [reply quickstart/101-analysis-services-create/TestRecord.md]: Fixed

File: quickstart/101-analysis-services-create/main.tf (hunk 3 above)
Comment (TomArcherMsft) [reply quickstart/101-analysis-services-create/main.tf:22]: Fixed.

File: quickstart/101-analysis-services-create/variables.tf (hunk 1 above)
Comment (TomArcherMsft) [reply quickstart/101-analysis-services-create/variables.tf]: Fixed

**Review by TomArcherMsft — COMMENTED** [review]
All changes made.

**Comment by TomArcherMsft** [comment]
@lonegunmanb This PR is ready for review/merge. Thanks.

File: quickstart/101-analysis-services-create/variables.tf [hunk 5]
```diff
@@ -0,0 +1,24 @@
+variable "resource_group_location" {
+  default     = "eastus"
```
Comment (grayzu) [inline quickstart/101-analysis-services-create/variables.tf:3]: These variables should define the type for the value. I think most of them are string but without the type defined, a user can pass in any type which is not correct.

**Review by grayzu — CHANGES_REQUESTED** [review]
Looks great! One minor comment.

File: quickstart/101-analysis-services-create/main.tf [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+  prefix = var.resource_group_name_prefix
+}
+
+resource "azurerm_resource_group" "rg" {
+  name     = random_pet.rg_name.id
+  location = var.resource_group_location
+}
+
+resource "azurerm_analysis_services_server" "server" {
+  name                = var.server_name
```
Comment (lonegunmanb) [inline quickstart/101-analysis-services-create/main.tf]: Could we add a random string as server name's prefix here to avoid potential naming conflict?

**Review by lonegunmanb — CHANGES_REQUESTED** [review]
@TomArcherMsft  The e2e test failed with following error message:

```text
azurerm_analysis_services_server.server: Still creating... [30s elapsed]
Test_Quickstarts/quickstart/101-analysis-services-create 2023-02-21T03:03:55Z command.go:185: ╷
Test_Quickstarts/quickstart/101-analysis-services-create 2023-02-21T03:03:55Z command.go:185: │ Error: creating Server (Subscription: "f7a632a5-49db-4c5e-9828-cd62cb753971"
Test_Quickstarts/quickstart/101-analysis-services-create 2023-02-21T03:03:55Z command.go:185: │ Resource Group Name: "rg-merry-ape"
Test_Quickstarts/quickstart/101-analysis-services-create 2023-02-21T03:03:55Z command.go:185: │ Server Name: "myserver"): performing Create: servers.ServersClient#Create: Failure sending request: StatusCode=0 -- Original Error: autorest/azure: Service returned an error. Status=<nil> Code="Conflict" Message="Server name 'myserver' is occupied." Details=[{"code":"RootActivityId","message":"c688c50f-ea6b-4e5c-adc0-cd51083278f6"},{"code":"Param1","message":"myserver"}]
Test_Quickstarts/quickstart/101-analysis-services-create 2023-02-21T03:03:55Z command.go:185: │ 
Test_Quickstarts/quickstart/101-analysis-services-create 2023-02-21T03:03:55Z command.go:185: │   with azurerm_analysis_services_server.server,
Test_Quickstarts/quickstart/101-analysis-services-create 2023-02-21T03:03:55Z command.go:185: │   on main.tf line 10, in resource "azurerm_analysis_services_server" "server":
Test_Quickstarts/quickstart/101-analysis-services-create 2023-02-21T03:03:55Z command.go:185: │   10: resource "azurerm_analysis_services_server" "server" {
Test_Quickstarts/quickstart/101-analysis-services-create 2023-02-21T03:03:55Z command.go:185: │ 
Test_Quickstarts/quickstart/101-analysis-services-create 2023-02-21T03:03:55Z command.go:185: ╵
```

I would suggest adding `random_string` as resource names' prefixes to avoid potential naming conflicts.

**Comment by TomArcherMsft** [comment]
@lonegunmanb @grayzu I believe I've made all the edits. Please have a look at how I defined the random_string for the analysis server name.

**Comment by grayzu** [comment]
lgtm

**Pull request merged**